### PR TITLE
Move gcs-overrides-enabled concept into RC_Channel

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by WickedShell** [comment]
Params for rover need to include the `RC Override Enable` option since it now works there as well. (Ideally there should be a generic macro we could include in the docs for all the options that are common which it could include, but we don't have that option at the moment).

You actually sort of need to chase the removal of `failsafe.rc_override_active`? (It's a small patch/addition). Otherwise if you toggle the switch and don't send new messages the copter state variable is still set to true. I was actually starting a PR that removed this when you sent this one in, but I'll just stash it for now.

**Comment by peterbarker** [comment]
@WickedShell chased failsafe.rc_override_active and fixed the parameters.

File: ArduCopter/radio.cpp [hunk 1]
```diff
@@ -115,7 +112,7 @@ void Copter::read_radio()
     }else{
         uint32_t elapsed = tnow_ms - last_radio_update_ms;
         // turn on throttle failsafe if no update from the RC Radio for 500ms or 2000ms if we are using RC_OVERRIDE
-        if (((!failsafe.rc_override_active && (elapsed >= FS_RADIO_TIMEOUT_MS)) || (failsafe.rc_override_active && (elapsed >= FS_RADIO_RC_OVERRIDE_TIMEOUT_MS))) &&
+        if (((!RC_Channels::has_active_overrides() && (elapsed >= FS_RADIO_TIMEOUT_MS)) || (RC_Channels::has_active_overrides() && (elapsed >= FS_RADIO_RC_OVERRIDE_TIMEOUT_MS))) &&
```
Comment (WickedShell) [inline ArduCopter/radio.cpp]: Probably worth stashing this in a local bool rather then calling it twice, it's a fairly expensive call (grab the singleton + iterates all the channels).

Comment (peterbarker) [reply ArduCopter/radio.cpp]: Realised we can not call has_active_overrides in the we've-got-radio case, too.  We only need to know if there are overrides when determining whether the RC receiver is present - which we've decided is always-present-if-ever-present.

**Comment by peterbarker** [comment]
Closed in favour of #9275

**Pull request closed without merging**